### PR TITLE
refactor: simplify how Frontend instance handles other protocols

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR is part of a series of changes to gradually adopt the Arrow Flight, and remove the old GRPC interface finally.

This PR refactors how Frontend instance handles the read/write of influxdb, opentsdb and prometheus protocols. Now they all go through the GRPC interface, greatly unify and simplify the Frontend instance either in standalone or distributed mode. 

Should be ready when #827 is merged.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
lgtm

File: src/common/substrait/src/types.rs [hunk 1]
```diff
@@ -151,10 +151,14 @@ pub(crate) fn scalar_value_as_literal_type(v: &ScalarValue) -> Result<LiteralTyp
             ScalarValue::Int64(Some(v)) => LiteralType::I64(*v),
             ScalarValue::LargeUtf8(Some(v)) => LiteralType::String(v.clone()),
             ScalarValue::LargeBinary(Some(v)) => LiteralType::Binary(v.clone()),
+            ScalarValue::TimestampMillisecond(Some(millis), _) => {
```
Comment (v0y4g3r) [inline src/common/substrait/src/types.rs:154]: Also convert timestamps with other units?

File: src/frontend/src/error.rs [hunk 2]
```diff
@@ -271,12 +271,6 @@ pub enum Error {
         source: common_grpc_expr::error::Error,
     },
 
-    #[snafu(display("Failed to deserialize insert batching: {}", source))]
```
Comment (v0y4g3r) [inline src/frontend/src/error.rs:274]: Looks like we can also delete these variants:
- `BumpTableId`
- `CreateDatabase`
- `CreateTableOnInsertion`
- `CatalogManager`
- `FullTableName`
- `ExecuteSql`
- `VectorComputation`

**Review by v0y4g3r — APPROVED** [review]
Finnaly we can get rid of matching mode :rofl:

**Review by Fengys123 — APPROVED** [review]
LGTM

**Pull request merged**